Gems/Profiler/Code/Source/CpuProfilerImpl.cpp

```diff
@@ -259,6 +290,29 @@ namespace Profiler
         }
     }
 
+    const char* CpuProfilerImpl::GetOrCreateFormatedString(const char* format, size_t argCount, va_list args)
+    {
+        AZStd::scoped_lock lock(m_dynamicNameMutex);
```

Comment:
Suggest a reader/writer lock pattern here so there's a fast path when we're just retrieving from the string pool